Sources/MarshaledObject.swift

```diff
@@ -130,6 +130,22 @@ public extension MarshaledObject {
         }
     }
 
+    public func value(for key: KeyType) throws -> [JSONObject] {
```

Comment:
I think I would prefer to see the `MarshalObject` typealias over `JSONObject` here. Although both type aliases are for convenience, I'd like to see the `JSONObject` typealias confined to the JSON helper code alone. This to reinforce the idea that Marshal is more than just JSON decoding.